lib/SparqlEndpointFetcher.ts

```diff
@@ -62,6 +62,45 @@ export class SparqlEndpointFetcher {
       ? (parsedQuery.queryType === 'DESCRIBE' ? 'CONSTRUCT' : parsedQuery.queryType) : "UNKNOWN";
   }
 
+  /**
+   * Get the query type of the given update query.
+   *
+   * This will parse the update query and thrown an exception on syntax errors.
+   *
+   * @param {string} query An update query.
+   * @return {'UNKNOWN' | {
+   *  insertDelete: ('insert' | 'delete' | 'deletewhere' | 'insertdelete')[];
+   *  management: ('copy' | 'move' | 'add' | 'load' | 'create' | 'clear' | 'drop')[]
+   * }} The included update operations.
+   */
+  public getUpdateTypes(query: string): 'UNKNOWN' | {
+    insertDelete: ('insert' | 'delete' | 'deletewhere' | 'insertdelete')[];
```

Comment:
Is there a reason why you chose to split the `insertDelete` and `management` operations?
I suspect it might be easier for libs downstream to have these in just a single datastructure.

Reply:
I did it primarily because it matched the separation in the type definition for sparqljs. In addition for our use case we need to add permissisons for the type of query users can perform which can usually be divided into `insertDelete` and `management` but this is not really a concern for this function.